src/masternodeman.cpp

```diff
@@ -1309,14 +1381,35 @@ bool CMasternodeMan::CheckMnbAndUpdateMasternodeList(CMasternodeBroadcast mnb, i
     LogPrint("masternode", "CMasternodeMan::CheckMnbAndUpdateMasternodeList -- masternode=%s\n", mnb.vin.prevout.ToStringShort());
 
     uint256 hash = mnb.GetHash();
-    if(mapSeenMasternodeBroadcast.count(hash)) { //seen
+    if(mapSeenMasternodeBroadcast.count(hash) && !mnb.fRecovery) { //seen
         LogPrint("masternode", "CMasternodeMan::CheckMnbAndUpdateMasternodeList -- masternode=%s seen\n", mnb.vin.prevout.ToStringShort());
         // less then 2 pings left before this MN goes into non-recoverable state, bump sync timeout
         if(GetTime() - mapSeenMasternodeBroadcast[hash].first > MASTERNODE_NEW_START_REQUIRED_SECONDS - MASTERNODE_MIN_MNP_SECONDS * 2) {
             LogPrint("masternode", "CMasternodeMan::CheckMnbAndUpdateMasternodeList -- masternode=%s seen update\n", mnb.vin.prevout.ToStringShort());
             mapSeenMasternodeBroadcast[hash].first = GetTime();
             masternodeSync.AddedMasternodeList();
         }
+        // did we ask this node for it?
+        if(pfrom && IsMnbRecoveryRequested(hash) && GetTime() < mMnbRecoveryRequests[hash].first) {
```

Comment:
I'm talking about an mnb from the network sent in response to a recovery query.  So fRecovery == false.  But this mnb is just a copy of an mnb that was already sent to the network so it is probably in mapSeenMasternodeBroadcast, so we probably won't enter the if block at line 1384 and the code at line 1393 won't be run.  So it looks like the if block after 1394 is basically unreachable since it only runs if IsMnbRecoveryRequested(hash).